sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusSenderAsyncClient.java

```diff
@@ -136,21 +136,24 @@ public String getEntityPath() {
     }
 
     /**
-     * Sends a message to a Service Bus queue or topic.
+     * Sends a set of messages to a Service Bus queue or topic using a batched approach. If the size of messages
+     * exceed the maximum size of a single batch, an exception will be triggered and the send will fail.
+     * By default, the message size is the max amount allowed on the link.
      *
-     * @param message Message to be sent to Service Bus queue or topic.
-     * @param sessionId the session id to associate with the message.
+     * @param messages Messages to be sent to Service Bus queue or topic.
      *
-     * @return A {@link Mono} the finishes this operation on service bus resource.
+     * @return A {@link Mono} that completes when all messages have been sent to the Service Bus resource.
      *
-     * @throws NullPointerException if {@code message} or {@code sessionId} is {@code null}.
+     * @throws NullPointerException if {@code messages} is {@code null}.
+     * @throws AmqpException if {@code messages} is larger than the maximum allowed size of a single batch.
      */
-    public Mono<Void> send(ServiceBusMessage message, String sessionId) {
-        Objects.requireNonNull(message, "'message' cannot be null.");
-        Objects.requireNonNull(sessionId, "'sessionId' cannot be null.");
+    public Mono<Void> send(Iterable<ServiceBusMessage> messages) {
+        Objects.requireNonNull(messages, "'messages' cannot be null.");
```

Comment:
This should return `monoError(logger, new NullPointerException("'messages' cannot be null.");` rather than throwing. There are other instances that I need to look at and fix. It's odd when async operations throw instead of returning the terminal error state.